src/remote.js

```diff
@@ -194,64 +202,75 @@
             );
           },
           function(error) {
-            util.err('Unable to create monaca project: ' + error);
+            util.err('Unable to create monaca project: ' + JSON.stringify(error));
             Q.reject(error);
           }
         )
         .then(
           function() {
-            // open the browser if no platform parameter is provided.
-            if (!argv.platform) {
-              var url = 'https://ide.monaca.mobi/project/' + projectInfo.projectId + '/' + (argv['build-type'] ? 'debugger' : 'build');
-              monaca.getSessionUrl(url)
-              .then(
-                function(url) {
-                  open(url);
-                },
-                function(error) {
-                  util.err('Unable to open build page.');
-                  process.exit(1);
-                }
-              );
-            }
-            else {
-              // Build project on Monaca Cloud and download it into ./build folder.
-              util.print("Building project on Monaca Cloud...");
-              monaca.buildProject(projectInfo.projectId, params).
-              then(function(result) {
-                if(result.binary_url) {
-                  return result.binary_url;
-                }
-                else {
-                  return Q.reject(result.error_message);
-                }
-              },
-              function(err) {
-                return Q.reject(err);
-              },
-              function (progress) {
-                util.print(progress);
-              })
-              .then(
-                function(url) {
-                  monaca.getSessionUrl(url).then(
-                    function(sessionUrl) {
-                      var buildDir = "";
-                      shell.mkdir('-p', path.join(cwd, 'build'));
-                      monaca.download(sessionUrl, {}, function(response) {
-                        var filename = "";
-                        if (typeof response.headers['content-disposition'] === 'string') {
-                          filename = response.headers['content-disposition'].match(/filename="?([^"]+)"?/)[1];
-                        }
-                        buildDir = path.join(cwd, "build", filename);
-                        return buildDir;
-                      }
-                      ).then(
-                        function(name) {
-                          util.print("Your package is stored at " + buildDir);
+            monaca.checkBuildAvailability(projectInfo.projectId, params.platform, params.purpose)
```

Comment:
You should return a `Promise`.
Do like:

```
return monaca.checkBuildAvailability(...)
```

Please read npm Q module for how to chain Promises.